src/swap/sideshift.js

```diff
@@ -0,0 +1,310 @@
+): Promise<any> {
+  const { fromCurrencyCode, toCurrencyCode, fromWallet, nativeAmount } = request
+  const { denominations, metaTokens } = fromWallet.currencyInfo
+  if (rate.error) {
+    throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+  }
+  const multiplier = denominations.find(d => d.name === fromCurrencyCode)
+    ? denominations.find(d => d.name === fromCurrencyCode).multiplier
+    : metaTokens
+        .find(t => t.currencyCode === fromCurrencyCode)
+        .denominations.find(d => d.name === fromCurrencyCode).multiplier
+
+  const nativeDepositAmount = mul(depositAmount, multiplier)
+
+  const amount =
+    request.quoteFor === 'from' ? nativeAmount : nativeDepositAmount
+
+  const nativeMin = await fromWallet.denominationToNative(
+    rate.min,
+    fromCurrencyCode
+  )
+
+  const nativeMax = await fromWallet.denominationToNative(
+    rate.max,
+    fromCurrencyCode
+  )
+
+  if (lt(amount, nativeMin)) {
+    throw new SwapBelowLimitError(swapInfo, nativeMin)
+  }
+
+  if (gt(amount, nativeMax)) {
+    throw new SwapAboveLimitError(swapInfo, nativeMax)
+  }
+}
+
+export function makeSideshiftPlugin(
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const { io, initOptions } = opts
+  const baseUrl = 'https://sideshift.ai/api/v1/'
+
+  async function get(path: string): Promise<any> {
+    const url = `${baseUrl}${path}`
+    const reply = await io.fetchCors(url)
+    return checkReplyForError(reply)
+  }
+
+  async function post(path, body): Promise<any> {
+    const url = `${baseUrl}${path}`
+    const reply = await io.fetchCors(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    })
+    return checkReplyForError(reply)
+  }
+
+  const out: EdgeSwapPlugin = {
+    swapInfo,
+    async fetchSwapQuote(request: EdgeSwapRequest): Promise<EdgeSwapQuote> {
+      const permission: Permission = await get('permissions')
+      if (
+        permission.createOrder === false ||
+        permission.createQuote === false
+      ) {
+        throw new SwapPermissionError(swapInfo, 'geoRestriction')
+      }
+
+      const [depositAddress, settleAddress] = await Promise.all([
+        getAddress(request.fromWallet, request.fromCurrencyCode),
+        getAddress(request.toWallet, request.toCurrencyCode)
+      ])
+
+      let safeFromCurrencyCode = request.fromCurrencyCode.toLowerCase()
+      let safeToCurrencyCode = request.toCurrencyCode.toLowerCase()
+      if (CURRENCY_CODE_TRANSCRIPTION[request.fromCurrencyCode]) {
+        safeFromCurrencyCode =
+          CURRENCY_CODE_TRANSCRIPTION[request.fromCurrencyCode]
+      }
+      if (CURRENCY_CODE_TRANSCRIPTION[request.toCurrencyCode]) {
+        safeToCurrencyCode = CURRENCY_CODE_TRANSCRIPTION[request.toCurrencyCode]
+      }
+
+      const rate: Rate = await get(
+        `pairs/${safeFromCurrencyCode}/${safeToCurrencyCode}`
+      )
+
+      const quoteAmount =
```

Comment:
Let's reduce use of `await` inside ternary operators

Reply:
Can do with a single `await`.

`await (request.codeFor === 'from' ? request.fromWallet.x() : request.toWallet.x())`.